src/classes/BDI_DataImportService.cls

```diff
@@ -305,31 +313,37 @@ global with sharing class BDI_DataImportService {
     /*******************************************************************************************************
     * @description whether the Account Custom Unique ID field is in our account matching rules
     */
-    private boolean isCustomIdInAccountMatchRules {
+    private Boolean isCustomIdInAccountMatchRules {
         get {
             String str = diSettings.Account_Custom_Unique_ID__c;
-            if (str == null || str == '')
+            if (str == null || str == '') {
```

Comment:
A better var name for this would be `customId`. Also, this could be simplified to `String.isEmpty(customId)`.